### ISSUE TITLE
Endless build loop with `fix` option

### ISSUE THREAD

**Issue by renarsvilnis** [webpack-contrib/stylelint-webpack-plugin#96]
Hey,
when using stylelint [`fix`](https://stylelint.io/user-guide/node-api/#fix) option it creates an endless build loop if errors are present. 

Stylelint tries to fix errors, but there are errors it can't fix and it writes the same contents of the file back, which retriggers a new build. I'm starting to think maybe I should submit this to `stylelint` instead of this repo. 

```javascript
plugins [
  new StyleLintPlugin({
    fix: false
  }),
  ...
]
```

**Comment by JaKXz** [comment]
@renarsvilnis thanks for reporting this. I think it should be discussed upstream in stylelint to see how non-correctable errors should be handled, but perhaps this plugin should also be able to detect this kind of infinite loop... 

I'm not sure where to start yet, and don't have the cycles to investigate ATM, so if you think you can put in a failing test then that would be a great start!

**Comment by renarsvilnis** [comment]
@JaKXz I'll look into how webpack plugins work and try to understand what's going into the plugin and what's coming out.  But the best result probably is to begin a discussion on stylelint.

**Comment by dkrnl** [comment]
I catch similar bug, temporary solution:
```js
const DEV_SERVER = path.basename(require.main.filename) === 'webpack-dev-server.js';
...
new StyleLintPlugin({
    fix: !DEV_SERVER,
}),
```

**Comment by benjixx** [comment]
@renarsvilnis Have you found a solution or workaround to this problem?

**Comment by benjixx** [comment]
Enabling the caching for stylelint seems to improve the situation a lot, although not all build loops are gone, e.g. when I add a blank line at the end of a CSS rule block I still run into a build loop.

**Comment by calledT** [comment]
I've create a [repo](https://github.com/calledT/webpack-stylelint-autofix-issue) to show this issue.
After `npm install` and exec `npm start`, it will run into endless status.

**Comment by ypconstante** [comment]
Any follow up in this behavior? In my situation, the loop happens even if there is no error in the file.

**Comment by JaKXz** [comment]
I would accept a PR that adds @dkrnl's workaround as the default for the `fix` option, but I'm not  a fan of that solution. Any deeper investigation is welcomed!

**Comment by JCJDesigns** [comment]
Also having this problem with webpack when using the `webpack --watch` command.

Here's my setup:

**/package.json**
```json
{
  "name": "LintTest",
  "version": "1.0.0",
  "main": "index.js",
  "scripts": {
    "build": "webpack -p",
    "watch": "webpack --watch"
  },
  "devDependencies": {
    "babel-core": "^6.26.0",
    "babel-loader": "^7.1.2",
    "babel-preset-env": "^1.6.1",
    "css-loader": "^0.28.8",
    "extract-text-webpack-plugin": "^3.0.2",
    "node-sass": "^4.7.2",
    "postcss-loader": "^2.0.10",
    "sass-loader": "^6.0.6",
    "stylelint": "^8.4.0",
    "stylelint-webpack-plugin": "^0.9.0",
    "webpack": "^3.10.0"
  },
  "browserslists": [
    ">= 1%",
    "last 20 versions"
  ]
}
```

**/webpack.config.js:**
```javascript
const path = require('path');
const ExtractTextPlugin = require('extract-text-webpack-plugin');
const StyleLintPlugin = require('stylelint-webpack-plugin');

const config = {
  entry: ['./src/js/index.js', './src/css/main.scss'],
  output: {
    path: path.resolve(__dirname, 'public'),
    filename: 'js/bundle.js'
  },
  module: {
    rules: [
      {
        test: /\.js$/,
        use: 'babel-loader'
      },
      {
        test: /\.scss$/,
        loader: ExtractTextPlugin.extract(['css-loader', 'postcss-loader', 'sass-loader'])
      }
    ]
  },
  plugins: [
    new ExtractTextPlugin({
      filename: 'css/styles.css',
      allChunks: true
    }),
    new StyleLintPlugin({
      configFile: '.stylelintrc',
      files: 'src/**/*.scss',
      syntax: 'scss',
      fix: true,
    })
  ]
};

module.exports = config;
```

**/.stylelintrc:**
```json
{
    "rules": {
        "indentation": 2,
        "string-quotes": "single",
        "color-hex-case": "lower",
        "color-hex-length": "short",
        "selector-no-qualifying-type": true,
        "selector-combinator-space-after": "always",
        "selector-attribute-quotes": "always",
        "selector-attribute-operator-space-before": "never",
        "selector-attribute-operator-space-after": "never",
        "selector-attribute-brackets-space-inside": "never",
        "declaration-block-trailing-semicolon": "always",
        "declaration-colon-space-before": "never",
        "declaration-colon-space-after": "always",
        "property-no-vendor-prefix": true,
        "value-no-vendor-prefix": true,
        "number-leading-zero": "always",
        "function-url-quotes": "always",
        "font-family-name-quotes": "always-where-required",
        "comment-whitespace-inside": "always",
        "at-rule-no-vendor-prefix": true,
        "selector-pseudo-element-colon-notation": "double",
        "selector-no-vendor-prefix": true,
        "media-feature-range-operator-space-before": "always",
        "media-feature-range-operator-space-after": "always",
        "media-feature-parentheses-space-inside": "never",
        "media-feature-name-no-vendor-prefix": true,
        "media-feature-colon-space-before": "never",
        "media-feature-colon-space-after": "always"
    }
}
```

**Comment by hartey11** [comment]
Hi, is there any progress on this? the fix `fix: path.basename(require.main.filename) !== 'webpack-dev-server.js',` doesn't work for me

**Comment by calledT** [comment]
This issus may due to [Files created right before watching starts make watching go into a loop](https://github.com/webpack/watchpack/issues/25), after add [time-fix-plugin](https://github.com/egoist/time-fix-plugin), it work well with `fix: true` config.

**Comment by ricardogobbosouza** [comment]
Upgrade to latest version, if problem persists reopen this
